bpf-prog/block-iptables/bpf/src/block_iptables.bpf.c

```diff
@@ -0,0 +1,181 @@
+//go:build ignore
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#define sk_family __sk_common.skc_family
+#define EPERM 1
+#define IPT_SO_SET_REPLACE 64
+#define TASK_COMM_LEN 16
+#define COMM_COUNT 3
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+volatile const u64 host_netns_inode = 4026531840; // Initialized by userspace
+
+struct {
+    __uint(type, BPF_MAP_TYPE_ARRAY);
```

Comment:
why this can't be percpu array?

Reply:
Discussed offline - percpu is useful when there is high contention, in this case a single array isn't a big performance problem with the level of concurrency we have.